app/helpers/social_accounts_helper.rb

```diff
@@ -0,0 +1,28 @@
+module SocialAccountsHelper
+  # rubocop:disable Metrics/CyclomaticComplexity
```

Comment:
What's the reason for this? IMHO we should either generally decide against this rule or follow it in every case. Otherwise, the code structure will become inconsistent.

Reply:
I would strongly advise against turning off whole Cops if we cannot fulfill them in one place. I would rather have some places marked as inconsistent than inconstient code all over our application.

Furthermore the following should be equivalent and RuboCop is not offended by it:

```ruby
LINK_MAP = {
    'GitHub' => ->(name) { "https://github.com/#{name}" },
    'Telegram' => ->(name) { "https://t.me/#{name}" },
    'Facebook' => ->(name) { "https://facebook.com/#{name}" },
    'Twitter' => ->(name) { "https://twitter.com/#{name}" },
    'GitLab@HPI' => ->(name) { "https://gitlab.hpi.de.com/#{name}" },
    'Slack' => proc { 'https://slack.com/' },
    'Discord' => ->(name) { "https://discordapp.com/#{name}" }
  }.freeze

  def generate_link(social_account)
    LINK_MAP[social_account.network].call(social_account.user_name)
  end
```

Reply:
Oh and the reason is that RuboCop does not like so many `if`s in one method :sweat_smile:

Reply:
You can try the code above by adding it and the following snippet to a local ruby file:

```ruby
class SocialNetwork
  attr_accessor :user_name, :network
end

sn = SocialNetwork.new()
sn.network = 'Slack'
sn.user_name = 'felix.auringer'

puts generate_link(sn)
```

Reply:
Ah yes, that refactoring was quite what I had proposed :-)

> I would strongly advise against turning off whole Cops if we cannot fulfill them in one place. I would rather have some places marked as inconsistent than inconstient code all over our application.

Agreed, but if we agree on using a linter, IMHO we should also reduce the occasions of manually overridden rules to an absolute minimum. Instead, we should refactor the reported places in code according to the linter output ...

Reply:
I agree, manually adjusting Cops inline should be one of the last opttions. And increasing the allowed cyclomatic complexity would also be an option but if it is really just one method that cannot be refactored properly I wouldn't relax the rules for all other methods too. But I think we have a possible refactoring here so that it shouldn't be necessary to touch RuboCop at all :tada: